src/nupic/algorithms/SpatialPooler.hpp

```diff
@@ -306,8 +308,8 @@ namespace nupic {
            */
           virtual void save(ostream& outStream) const;
 
-          virtual void write(ostream& stream) const;
-          virtual void write(SpatialPoolerProto::Builder& proto) const;
+          using Serializable::write;
```

Comment:
Because otherwise you can't do this:

    SpatialPooler sp = new SpatialPooler...
    // define stream
    sp.write(stream);